pennylane/devices/default_clifford.py

```diff
@@ -0,0 +1,726 @@
+                f"Currently this device doesn't support executing non-Clifford operations, got max_error={max_error}."
+            )
+
+        if state in ["tableau", "state_vector"]:
+            self._state = state
+        else:
+            raise ValueError(
+                f"Keyword state only accepts two options: 'tableau' and 'state_vector', got {state}."
+            )
+
+        seed = np.random.randint(0, high=10000000) if seed == "global" else seed
+        if qml.math.get_interface(seed) == "jax":  # pragma: no cover
+            self._prng_key = seed
+            self._rng = np.random.default_rng(None)
+        else:
+            self._prng_key = None
+            self._rng = np.random.default_rng(seed)
+        self._debugger = None
+
+    def _setup_execution_config(self, execution_config: ExecutionConfig) -> ExecutionConfig:
+        """This is a private helper for ``preprocess`` that sets up the execution config.
+
+        Args:
+            execution_config (ExecutionConfig)
+
+        Returns:
+            ExecutionConfig: a preprocessed execution config
+
+        """
+
+        updated_values = {}
+        if execution_config.gradient_method == "best":  # pragma: no cover
+            updated_values["gradient_method"] = None
+        updated_values["use_device_gradient"] = True
+        updated_values["use_device_jacobian_product"] = False
+        if execution_config.grad_on_execution is None:
+            updated_values["grad_on_execution"] = False
+        updated_values["device_options"] = dict(execution_config.device_options)  # copy
+        if "max_workers" not in updated_values["device_options"]:
+            updated_values["device_options"]["max_workers"] = self._max_workers
+        if "rng" not in updated_values["device_options"]:
+            updated_values["device_options"]["rng"] = self._rng
+        if "prng_key" not in updated_values["device_options"]:
+            updated_values["device_options"]["prng_key"] = self._prng_key
+        return replace(execution_config, **updated_values)
+
+    def preprocess(
+        self,
+        execution_config: ExecutionConfig = DefaultExecutionConfig,
+    ) -> Tuple[TransformProgram, ExecutionConfig]:
+        """This function defines the device transform program to be applied and an updated device configuration.
+
+        Args:
+            execution_config (Union[ExecutionConfig, Sequence[ExecutionConfig]]): A data structure describing the
+                parameters needed to fully describe the execution.
+
+        Returns:
+            TransformProgram, ExecutionConfig: A transform program that when called returns QuantumTapes that the device
+            can natively execute as well as a postprocessing function to be called after execution, and a configuration with
+            unset specifications filled in.
+
+        This device:
+
+        * Supports any qubit operations that provide a matrix
+        * Currently does not support finite shots
```

Comment:
Is this going to be added in a subsequent PR?

Reply:
Yes, I will be adding support for shots in the next PR.